### PR TITLE
fix: return coordinates to its original scale when using stitcher

### PR DESCRIPTION
**Pull request by lgemc**
Hello, first of all, thanks for making this repo open source, I have learned a lot from it 🙏

When I ran the testing pipeline, I noticed that detections weren’t properly calculated. This happens because of a small bug in the detection point calculation, which doesn’t take into account the stitcher’s original downsampling. In the end, points must be upscaled to fit the original image size when a stitcher is used.

Feel free to share any comments about things I should change to get the fix right. Once again, thanks for this useful and instructive repo 🙏

I attach two images with the before and after

image

image

Also the code used on the evaluator when testing the solution and the error:

```python
down_ratio = 2
test_dataset = CSVDataset(
    csv_file = f'{ROOT_DIR}/val_patches/gt.csv',
    root_dir = f'{ROOT_DIR}/val_patches',
    albu_transforms = [A.Normalize(p=1.0)],
    end_transforms = [DownSample(down_ratio=down_ratio, anno_type='point')]
    )

test_dataloader = DataLoader(dataset = test_dataset, batch_size = 1, shuffle = False, num_workers=8)

test_evaluator = HerdNetEvaluator(
    model=herdnet,
    dataloader=test_dataloader,
    metrics=metrics,
    stitcher=stitcher,
    work_dir=test_dir,
    header='test'
    )
    
test_f1_score = test_evaluator.evaluate(returns='f1_score')

test_evaluator.detections.to_csv("test_evaluator_detections.csv")
```

### CODE REVIEW

**Comment by Alexandre-Delplanque** [comment]
Hi @lgemc,

Thanks for pointing that out! 

Stitcher classes have an arg called [`up`](https://github.com/Alexandre-Delplanque/HerdNet/blob/7e25f482d875522c59c446dc0c78c8f6f2dd448d/animaloc/eval/stitchers.py#L71) to specify wether you want to upscale the predicted map or not, using the `down_ratio` value.

Did you first try to set `up=True`?

Best,

Alexandre

**Comment by lgemc** [comment]
Hello, thanks for your response, i tried using `up=True`, but it broke the metric calculation is broken

```
test [  1/258] eta: 0:13:16 n: 1 recall: 0.0 precision: 0.0 f1-score: 0.0 MAE: 4.0 MSE: 16.0 RMSE: 4.0 time: 3.0864 data: 0.2682 max mem: 2730
```

But nope, current pr will broke the inference script, maybe is there another way to have the stitcher working in both, evaluator final detections and internal f1 score calculation...

**Comment by Alexandre-Delplanque** [comment]
Hi @lgemc ,

Thanks for your feedback, do not forget to set `down_ratio=1` in your test dataset's `end_transforms` whe you're using `up=True`. Otherwise, the `Metrics` will compare down sampled ground truth points with upscaled detections.

Let me know if it works.

Best,

Alexandre
